### PR TITLE
Add strncpy() conformance tests and assertion.xml

### PR DESCRIPTION
**Pull request by padmashree9110**
Adds
1> 1-1, 2-1 tests
2> assertions.xml
3> coverage.txt

Change-Id: Id21aa9c6e03ac94044ed094cf4ddcf3651fe73ad
Signed-off-by: Mandri, Padmashree <padmashree.mandri@intel.com>
Signed-off-by: Palleti, Avinash Reddy <avinash.reddy.palleti@intel.com>

### CODE REVIEW

**Comment by anitha-suresh** [comment]
@metan-ucw : Could you please review/merge the patch.

**Comment by padmashree9110** [comment]
@ngie-eign Can you please review this.

**Comment by padmashree9110** [comment]
@ngie-eign : Upadted PR

File: testcases/open_posix_testsuite/conformance/interfaces/strncpy/2-1.c [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+char *random_string(int len)
+{
+    int i;
+    char *output_string;
+    output_string = malloc(len + 1);
+    if (output_string == NULL) {
+        printf(TNAME " Failed to allocate memory\n");
+        exit(PTS_UNRESOLVED);
+    }
+    for (i = 0; i < len; i++)
```
Comment (ngie-eign) [inline testcases/open_posix_testsuite/conformance/interfaces/strncpy/2-1.c]: Please add enclosing braces for clarity.

File: testcases/open_posix_testsuite/conformance/interfaces/strncpy/2-1.c [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        sample_str_2[i + EXTRA_BYTES] = '\0';
+
+        if (strcmp(sample_str_1, sample_str_2) != 0) {
+            printf(TNAME " Test failed, string copy failed. "
+                        "Expected string: %s, But got: %s\n"
+                        , sample_str_1, sample_str_2);
+            exit(PTS_UNRESOLVED);
+        } else if (ret_str != sample_str_2) {
+            printf(TNAME " Test Failed, return is not as expected. "
+                        "Expected return: %p, But obtained: %p\n"
```
Comment (ngie-eign) [inline testcases/open_posix_testsuite/conformance/interfaces/strncpy/2-1.c]: Please move the comma before `sample_str_2`.

File: testcases/open_posix_testsuite/conformance/interfaces/strncpy/2-1.c [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            exit(PTS_UNRESOLVED);
+        }
+        sample_str_1 = random_string(i);
+        ret_str = strncpy(sample_str_2, sample_str_1, i + EXTRA_BYTES);
+        sample_str_2[i + EXTRA_BYTES] = '\0';
+
+        if (strcmp(sample_str_1, sample_str_2) != 0) {
+            printf(TNAME " Test failed, string copy failed. "
+                        "Expected string: %s, But got: %s\n"
+                        , sample_str_1, sample_str_2);
```
Comment (ngie-eign) [inline testcases/open_posix_testsuite/conformance/interfaces/strncpy/2-1.c]: Please move the comma before `sample_str_1`.

**Comment by ngie-eign** [comment]
Please fix the style issues in another PR, along with the ones I brought up in https://github.com/linux-test-project/ltp/pull/384 .

**Pull request merged**